### PR TITLE
feat: Add Save Return to Compliance Pages

### PR DESCRIPTION
**Pull request by dhaselhan**
* Add a save and return button beneath all tables in Compliance Reporting
* Expand BCGrid to support this functionality since validity of rows impacts how it works
* Remove Callable Model, looks  to be an old dead modal built using bootstrap
* Fix incorrect route constant
* Remove deprecated reactiveCustomComponents attribute

Closes: https://github.com/bcgov/lcfs/issues/848

### CODE REVIEW

File: frontend/src/components/BCDataGrid/BCGridBase.jsx [hunk 1]
```diff
@@ -33,7 +33,6 @@ export const BCGridBase = forwardRef(({ autoSizeStrategy, ...props }, ref) => {
         suppressDragLeaveHidesColumns
         suppressMovableColumns
         suppressColumnMoveAnimation={false}
-        reactiveCustomComponents
```
Comment (AlexZorkin) [inline frontend/src/components/BCDataGrid/BCGridBase.jsx:36]: We have to keep this option to allow our inputs to commit values. Def not ideal but until we find a work around lets leave this here.

**Review by AlexZorkin — CHANGES_REQUESTED** [review]
Looks good, just one minor change to include the reactiveCustomComponents option. Please also merge release into this branch again. ty

File: frontend/src/components/BCDataGrid/BCGridBase.jsx (hunk 1 above)
Comment (dhaselhan) [reply frontend/src/components/BCDataGrid/BCGridBase.jsx:36]: That's odd, its marked as deprecated in their documentation as default on. Are we on an older version?

Comment (AlexZorkin) [reply frontend/src/components/BCDataGrid/BCGridBase.jsx:36]: we're running v32.0.2 and there is a v32.1.0
I have tested without this option and the inputs break so we'll have to investigate further. For this ticket lets just leave it in for now.

Comment (dhaselhan) [reply frontend/src/components/BCDataGrid/BCGridBase.jsx]: Done, added back

**Review by AlexZorkin — APPROVED** [review]

**Pull request merged**